instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/snippet/InjectionState.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.servlet.v3_0.snippet;
+
+// this is shared by both ServletOutputStream and PrintWriter injection
+public class InjectionState {
+  private static final int HEAD_TAG_WRITTEN_FAKE_VALUE = -1;
+  private static final int HEAD_TAG_LENGTH = "<head>".length();
```

Comment:
Idea for improvement in future PRs: `head` can have attributes, so it's possible that for some HTML documents this will never match

Reply:
// future improvements adding